### PR TITLE
Added not found error if no partition assignment on seek

### LINKED ISSUE #195: Handle error on seek operations if the consumer is not assigned to the partitions
What happens if the consumer does a seek operation specifying offsets but for partitions to which it isn't subscribed (or they are not assigned)?
The same for seek to end and beginning.

### PR DESCRIPTION
**Pull request by ppatierno**
This PR fixes #195

### CODE REVIEW

File: src/main/java/io/strimzi/kafka/bridge/http/HttpSinkBridgeEndpoint.java [hunk 1]
```diff
@@ -130,11 +130,15 @@ private void doSeek(JsonObject bodyAsJson) {
             if (done.succeeded()) {
                 HttpUtils.sendResponse(routingContext.response(), HttpResponseStatus.NO_CONTENT.code(), null, null);
             } else {
+                HttpResponseStatus statusCode = HttpResponseStatus.INTERNAL_SERVER_ERROR;
+                if (done.cause() instanceof IllegalStateException) {
```
Comment (tombentley) [inline src/main/java/io/strimzi/kafka/bridge/http/HttpSinkBridgeEndpoint.java]: I'm surprised that the Apache `KafkaConsumer` doesn't have a dedicated exception for this. We can't really be 100% sure that all ISEs from `seek()` have as their cause the reasons we expect here. But matching on the exception message seems fragile and what other choice is there? So I'm OK with this, but have some qualms about it.

File: src/main/resources/openapi.json [hunk 2]
```diff
@@ -130,7 +130,7 @@
                         "description": "Seek to the beginning performed successfully."
                     },
                     "404": {
-                        "description": "The specified consumer instance was not found.",
+                        "description": "The specified consumer instance or the partition assignment was not found.",
```
Comment (tombentley) [inline src/main/resources/openapi.json]:
```suggestion
                        "description": "The specified consumer instance was not found, or the specified consumer instance did not have one of the specified partitions assigned.",
```

File: src/main/resources/openapi.json [hunk 3]
```diff
@@ -194,7 +194,7 @@
                         "description": "Seek to the end performed successfully."
                     },
                     "404": {
-                        "description": "The specified consumer instance was not found.",
+                        "description": "The specified consumer instance or the partition assignment was not found.",
```
Comment (tombentley) [inline src/main/resources/openapi.json]:
```suggestion
                        "description": "The specified consumer instance was not found, or the specified consumer instance did not have one of the specified partitions assigned.",
```

File: src/main/resources/openapi.json [hunk 4]
```diff
@@ -387,7 +387,7 @@
                         "description": "Seek performed successfully."
                     },
                     "404": {
-                        "description": "The specified consumer instance was not found.",
+                        "description": "The specified consumer instance or the partition assignment was not found.",
```
Comment (tombentley) [inline src/main/resources/openapi.json]:
```suggestion
                        "description": "The specified consumer instance was not found, or the specified consumer instance did not have one of the specified partitions assigned.",
```

File: src/main/resources/openapiv2.json [hunk 5]
```diff
@@ -128,7 +128,7 @@
             "description": "Seek to the beginning performed successfully."
           },
           "404": {
-            "description": "The specified consumer instance was not found.",
+            "description": "The specified consumer instance or the partition assignment was not found.",
```
Comment (tombentley) [inline src/main/resources/openapiv2.json]:
```suggestion
            "description": "The specified consumer instance was not found, or the specified consumer instance did not have one of the specified partitions assigned.",
```

File: src/main/resources/openapiv2.json [hunk 6]
```diff
@@ -188,7 +188,7 @@
             "description": "Seek to the end performed successfully."
           },
           "404": {
-            "description": "The specified consumer instance was not found.",
+            "description": "The specified consumer instance or the partition assignment was not found.",
```
Comment (tombentley) [inline src/main/resources/openapiv2.json]:
```suggestion
            "description": "The specified consumer instance was not found, or the specified consumer instance did not have one of the specified partitions assigned.",
```

File: src/main/resources/openapiv2.json [hunk 7]
```diff
@@ -355,7 +355,7 @@
             "description": "Seek performed successfully."
           },
           "404": {
-            "description": "The specified consumer instance was not found.",
+            "description": "The specified consumer instance or the partition assignment was not found.",
```
Comment (tombentley) [inline src/main/resources/openapiv2.json]:
```suggestion
            "description": "The specified consumer instance was not found, or the specified consumer instance did not have one of the specified partitions assigned.",
```

File: src/main/java/io/strimzi/kafka/bridge/http/HttpSinkBridgeEndpoint.java (hunk 1 above)
Comment (ppatierno) [reply src/main/java/io/strimzi/kafka/bridge/http/HttpSinkBridgeEndpoint.java]: I agree Tom ... but this is what we have from the native client right now :-(

**Pull request merged**